src/ast/variables/Variable.ts

```diff
@@ -12,7 +12,7 @@ import { LiteralValueOrUnknown, UnknownValue, UNKNOWN_EXPRESSION } from '../valu
 
 export default class Variable implements ExpressionEntity {
 	alwaysRendered = false;
-	exportName: string | null = null;
+	exportName: string[] | null = null;
```

Comment:
Please definitely rename it to `exportNames`. An array that does not have a plural `s` is very confusing when encountered elsewhere. Also I see you do a lot of checks like `if (exportNames && exportNames[0]) ...`. Depending on what the logic is, one of the checks will always be superfluous: If a variable is exported, `exportNames` needs to have at least length `1`.

My suggestion would be to change this to `exportNames: string[] = []` where an empty array represents that something is not exported. Then you can rely on this property being present and just need to query `.exportNames.length` to see if something is exported. Just make sure that these arrays are not mutated but replaced as a whole because these `Variable` instances are shared between outputs, there should be no interaction between outputs.